redis/spring-cloud-starter-stream-sink-redis/src/main/java/org/springframework/cloud/stream/app/redis/sink/RedisSinkProperties.java

```diff
@@ -79,51 +79,63 @@ public void setTopic(String topic) {
 		this.topic = topic;
 	}
 
-	public Expression getKeyExpression() {
-		return key != null ? new LiteralExpression(key) : parser.parseExpression(keyExpression);
+	public Expression keyExpression() {
```

Comment:
We still have the getter methods (down below) which return the exact value of the `Expression` while these simple methods are used to construct the expression based on the values from `String` or `Expression`.